### DISCUSSION TITLE
What does the same value for Redelivered and Deliver (manual ack) mean?

### DISCUSSION THREAD

**Discussion by ponponon** [rabbitmq/rabbitmq-server#10235] [Questions] [answered]
<img width="1353" alt="图片" src="https://github.com/rabbitmq/rabbitmq-server/assets/38725104/37db69fd-e985-4c0c-801d-6591420cbd55">

What does the same value for Redelivered and Deliver (manual ack) mean?

What does Deliver (manual ack) mean? Is it the speed of delivering messages to the rabbitmq server as a producer? Or is it the rate at which you consume messages as a consumer? If it is the former, what is the difference between the meaning of this value and publish?

What does ReDeliver mean? Does it mean repeat consumption?

**Comment by michaelklishin** [comment, marked as answer]
Please [consult this doc guide](https://rabbitmq.com/confirms.html).

"Delivery" is a delivery to a consumer. Redelivery means a message was requeued and is delivered for the 2nd, 3rd, …, Nth time (to another consumer or the same one). When this may happen is documented in the guide above.
